crates/turbopath/src/relative_unix_path_buf.rs

```diff
@@ -1,75 +1,163 @@
-use std::path::PathBuf;
+use std::{fmt::Debug, io::Write};
 
-use serde::Serialize;
+use bstr::{BString, ByteSlice};
 
-use crate::{IntoUnix, PathValidationError};
+use crate::{not_relative_error, PathError, PathValidationError};
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Serialize)]
-pub struct RelativeUnixPathBuf(PathBuf);
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+pub struct RelativeUnixPathBuf(BString);
 
 impl RelativeUnixPathBuf {
-    /// Create a new RelativeUnixPathBuf from a PathBuf by calling `into_unix()`
-    ///
-    /// NOTE: `into_unix` *only* converts Windows paths to Unix paths *on* a
-    /// Windows system. Do not pass a Windows path on a Unix system and
-    /// assume it'll be converted.
-    ///
-    /// # Arguments
-    ///
-    /// * `path`:
-    ///
-    /// returns: Result<RelativeUnixPathBuf, PathValidationError>
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// ```
-    pub fn new(path: impl Into<PathBuf>) -> Result<Self, PathValidationError> {
-        let path = path.into();
-        if path.is_absolute() {
-            return Err(PathValidationError::NotRelative(path));
+    pub fn new(path: impl Into<Vec<u8>>) -> Result<Self, PathError> {
+        let bytes: Vec<u8> = path.into();
+        if !bytes.is_empty() && bytes[0] == b'/' {
```

Comment:
nit: you could also do `bytes.first() == Some(b'/')`

Reply:
`bytes.first() == Some(&b'/')`, but yeah, I like that better.